### PR TITLE
📖  fix escaped html in markdown code segments

### PR DESCRIPTION
**Pull request by typeid**
**Issue #3534**
- we are using `markerdocs` as [mdBook preprocessor](https://rust-lang.github.io/mdBook/for_developers/preprocessors.html)  to generate most of our `References` section (e.g. [CRD Validation](https://book.kubebuilder.io/reference/markers/crd-validation))
- this preprocessor writes HTML but the text fields also contain markdown (e.g. [this line](https://github.com/kubernetes-sigs/controller-tools/blob/498444bd78eb90dbdf228dfd9ad333d68fe354e7/pkg/crd/markers/validation.go#L220))
- we call `html.EscapeString()` on the output of the preprocessor (see [here](https://github.com/kubernetes-sigs/kubebuilder/blob/68abac1dfb5550f7159dec4ce600cd9b6db925bb/docs/book/utils/markerdocs/html.go#L44))
- the preprocessed output is then parsed as markdown by mdBook

Basically, mdBook handles the content as raw strings (code block) thus quotation marks and other HTML escaped symbols are not escaped

1. Raw text:
```
 object: `{policy: "delete"}`
```

2. Preprocessor wraps it with HTML in escaped format

<pre>
object: `{policy: &#34;delete&#34;}`
</pre>

3. mdBook picks up on the markdown backticks and further wraps it in the according HTML for the markdown format without interpreting the escaped characters:
```
<code class="hljs">{policy: &#34;delete&#34;}</code>
```

This PR aims to fix this issue by not HTML escaping the markdown code segments contained in the text, leaving the behavior up to mdBook.

Fixes https://github.com/kubernetes-sigs/kubebuilder/issues/3540

### CODE REVIEW

**Comment by typeid** [comment]
`./install-and-build.sh` works locally. 
I saw that there was a fix on the `netlify/kubebuilder/deploy-preview`, and the deploy preview works on another PR. So I'm wondering if this error could be sporadic. Could we re-trigger the netlify job?

**Comment by typeid** [comment]
@camilamacedo86 Ah indeed, forgot to rebase. Done and the deploy-preview works now :)

**Review by camilamacedo86 — APPROVED** [review]
Thank you for the amazing contribution 🥇 

It is possible to check that it is fixed now:

Screenshot 2023-08-18 at 07 02 17

/approved

/lgtm

/ok-to-test
